nxdrive/exceptions.py

```diff
@@ -158,6 +158,12 @@ class UploadPaused(TransferPaused):
     pass
 
 
+class UploadCancelled(DriveError):
```

Comment:
You should crzate a parent class `TransferCancelled` and inherit from it. Like it is done for  `UploadPaused`.